restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java

```diff
@@ -2694,9 +2697,226 @@ else if (CollectionResult.class.isAssignableFrom(returnType))
+                + "the parameter with name '" + parameter.getName() + "'");
+          }
+
+          fieldNames.remove(field.getName());
+          break;
+        case CONTEXT:
+        case PAGING_CONTEXT_PARAM:
+          PagingContext defaultContext = (PagingContext) parameter.getDefaultValue();
+          RecordDataSchema.Field startField = criteriaSchema.getField(RestConstants.START_PARAM);
+          if (startField == null || startField.getType().getType() != DataSchema.Type.INT)
+          {
+            throw new ResourceConfigException("Linked @BatchFinder method '" + linkedBatchFinderName
+                + "' from @Finder Method named '" + finder.getFinderName()
+                + "' on class '" + resourceModel.getResourceClass().getName()
+                + "' has a @PagingContext or @Context param but no int '" + RestConstants.START_PARAM
+                + "' field in the criteria object");
+          }
+
+          if (startField.getDefault() == null || !startField.getDefault().equals(defaultContext.getStart()))
+          {
+            throw new ResourceConfigException("Linked @BatchFinder method '" + linkedBatchFinderName
+                + "' from @Finder Method named '" + finder.getFinderName()
+                + "' on class '" + resourceModel.getResourceClass().getName()
+                + "' has a @PagingContext or @Context param but the '" + RestConstants.START_PARAM
+                + "' field default values don't match");
+          }
+
+          RecordDataSchema.Field countField = criteriaSchema.getField(RestConstants.COUNT_PARAM);
+          if (countField == null || countField.getType().getType() != DataSchema.Type.INT)
+          {
+            throw new ResourceConfigException("Linked @BatchFinder method '" + linkedBatchFinderName
+                + "' from @Finder Method named '" + finder.getFinderName()
+                + "' on class '" + resourceModel.getResourceClass().getName()
+                + "' has a @PagingContext or @Context param but no int '" + RestConstants.COUNT_PARAM
+                + "' field in the criteria object");
+          }
+
+          if (countField.getDefault() == null || !countField.getDefault().equals(defaultContext.getCount()))
+          {
+            throw new ResourceConfigException("Linked @BatchFinder method '" + linkedBatchFinderName
+                + "' from @Finder Method named '" + finder.getFinderName()
+                + "' on class '" + resourceModel.getResourceClass().getName()
+                + "' has a @PagingContext or @Context param but the '" + RestConstants.COUNT_PARAM
+                + "' field default values don't match");
+          }
+
+          fieldNames.remove(RestConstants.START_PARAM);
+          fieldNames.remove(RestConstants.COUNT_PARAM);
+          break;
+        case PROJECTION:
+        case PROJECTION_PARAM:
+          RecordDataSchema.Field fieldsField = criteriaSchema.getField(RestConstants.FIELDS_PARAM);
```

Comment:
This is different from my understanding of batchFinder, paging should be specified as a separate parameter in BatchFinder instead of putting inside Criteria schema. See https://github.com/linkedin/rest.li/blob/master/restli-int-test-server/src/main/java/com/linkedin/restli/examples/greetings/server/BatchFinderResource.java#L54

Reply:
Ok fair point. If we do it this way then we will lose what the framework gives us. Simplified and changed. We will change batching logic accordingly.